evalml/pipelines/components/transformers/scalers/standard_scaler.py

```diff
@@ -41,15 +68,20 @@ def transform(self, X, y=None):
         """
         X = infer_feature_types(X)
         X = X.ww.select(exclude=["datetime"])
```

Comment:
I think dropping datetime columns like this is weird but we're doing this to keep this compatible with partial dependence. Just going to leave this here for now but I do the same [here](https://github.com/alteryx/evalml/pull/3686/files#diff-da2d6fc3e28ef107b8015b493e115ed8b6ed81d796782a174111d222b9d82ba7R50) and removed the separate dropping in `fit_transform()`